lib/widgets/lightbox.dart

```diff
@@ -208,11 +213,244 @@ class _LightboxPageState extends State<_LightboxPage> {
   }
 }
 
+class VideoLightboxPage extends StatefulWidget {
+  const VideoLightboxPage({
+    super.key,
+    required this.routeEntranceAnimation,
+    required this.message,
+    required this.src,
+  });
+
+  final Animation routeEntranceAnimation;
+  final Message message;
+  final Uri src;
+
+  @override
+  State<VideoLightboxPage> createState() => _VideoLightboxPageState();
+}
+
+class _VideoLightboxPageState extends State<VideoLightboxPage> {
+  // TODO(#38): Animate entrance/exit of header and footer
+  bool _headerFooterVisible = false;
+
+  VideoPlayerController? _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.routeEntranceAnimation.addStatusListener(_handleRouteEntranceAnimationStatusChange);
+    // We delay initialization by a single frame to make sure the
+    // BuildContext is a valid context, as its needed to retrieve
+    // the PerAccountStore & ZulipLocalizations during initialization.
+    SchedulerBinding.instance.addPostFrameCallback((_) => _initialize());
+  }
+
+  Future<void> _initialize() async {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    assert(debugLog('VideoPlayerController.networkUrl(${widget.src})'));
+    _controller = VideoPlayerController.networkUrl(widget.src, httpHeaders: {
+      if (widget.src.origin == store.account.realmUrl.origin) ...authHeader(
+        email: store.account.email,
+        apiKey: store.account.apiKey,
+      ),
+      ...userAgentHeader()
+    });
+    _controller!.addListener(_handleVideoControllerUpdates);
+
+    try {
+      await _controller!.initialize();
+      await _controller!.play();
+    } catch (error) { // TODO(log)
+      assert(debugLog("VideoPlayerController.initialize failed: $error"));
+      if (mounted) {
+        await showDialog(
+          context: context,
+          barrierDismissible: false,
```

Comment:
I'd also be interested in reunifying this with the methods in `dialog.dart`. Those are helpful for keeping UI consistency across our dialog interactions.

What's the reasoning for setting `barrierDismissible: false` here? It looks like that, and the logic to pop the lightbox in addition to the dialog, are the two differences from `showErrorDialog`.

We could add an option there to control these differences, given a clear story for what they're about. (Perhaps just a boolean to choose between the existing behavior and this behavior? Or perhaps the two changes shouldn't be coupled. It depends on what the reasoning is for this `barrierDismissible` change.)

Reply:
Added a comment on the reasoning for `barrierDismissible: false`.

```dart
// To avoid showing the disabled video lightbox for the unnsupported
// video, we make sure user doesn't reach there by dismissing the dialog
// by clicking around it, user must press the 'OK' button, which will
// take user back to content message list.
barrierDismissible: false,
```